client/nginx_test.go

```diff
@@ -224,9 +235,227 @@ func TestStreamDetermineUpdates(t *testing.T) {
 	}}
 
 	for _, test := range tests {
-		toAdd, toDelete := determineStreamUpdates(test.updated, test.nginx)
+		toAdd, toDelete, _ := determineStreamUpdates(test.updated, test.nginx)
 		if !reflect.DeepEqual(toAdd, test.expectedToAdd) || !reflect.DeepEqual(toDelete, test.expectedToDelete) {
 			t.Errorf("determiteUpdates(%v, %v) = (%v, %v)", test.updated, test.nginx, toAdd, toDelete)
 		}
 	}
 }
+
+func TestDetermineUpdatesWithUpdate(t *testing.T) {
+	maxConns := 1
+	var tests = []struct {
+		updated          []UpstreamServer
+		nginx            []UpstreamServer
+		expectedToUpdate []UpstreamServer
+	}{{
+		updated: []UpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		},
+		nginx: []UpstreamServer{
+			{
+				Server: "10.0.0.1:80",
+			},
+			{
+				ID:     2,
+				Server: "10.0.0.2:80",
+			},
+		},
+		expectedToUpdate: []UpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		}}}
+
+	for _, test := range tests {
+		_, _, toUpdate := determineUpdates(test.updated, test.nginx)
+		if !reflect.DeepEqual(toUpdate, test.expectedToUpdate) {
+			t.Errorf("determiteUpdates(%v, %v) = (%v)", test.updated, test.nginx, toUpdate)
+		}
+	}
+}
+
+func TestStreamDetermineUpdatesWithUpdate(t *testing.T) {
```

Comment:
```suggestion
func TestStreamDetermineUpdates(t *testing.T) {
```

Reply:
On second look, not sure my suggestion makes sense. The test calls `determineStreamUpdates` which returns `toCreate`, `toDelete` and `toUpdate` but in this test we just test `toUpdate`.

Instead maybe you can take a look to see if these cases can be tested as additional cases in the already existing `TestDetermineUpdates`?

Reply:
On second look, not sure my suggestion makes sense. The test calls `determineStreamUpdates` which returns `toCreate`, `toDelete` and `toUpdate` but in this test we just test `toUpdate`.

Instead maybe you can take a look to see if these cases can be tested as additional cases in the already existing `TestDetermineUpdates`?

Reply:
Hmm, not sure this change makes sense. See: https://github.com/nginxinc/nginx-plus-go-client/pull/34#discussion_r340247474